my_compassion/controllers/my2_donations.py

```diff
@@ -202,6 +202,36 @@ def remove_item(self, **post):
 
         return {"html": html_content}
 
+    @http.route(
+        "/my2/gifts/thankyou",
+        type="http",
+        auth="user",
+        website=True,
+        sitemap=False,
+    )
+    def my2_gifts_thank_you_page(self, **kwargs):
+        sale_order_id = int(kwargs.get("sale_order_id", 0))
+        if not sale_order_id:
+            raise NotFound("Sale order ID has not been found")
+
+        sale_order = (
+            request.env["sale.order"]
+            .sudo()
+            .search(
+                [
+                    ("id", "=", sale_order_id),
+                    ("partner_id", "=", request.env.user.partner_id.id),
+                ]
+            )
+        )
```

Comment:
We can avoid relying on query parameters.

See https://github.com/odoo/odoo/blob/cc0060e889603eb2e47fa44a8a22a70d7d784185/addons/website_sale/controllers/main.py#L1116-L1130.

```suggestion
        sale_order_id = request.session.get('sale_last_order_id')
        if sale_order_id:
            sale_order = request.env['sale.order'].sudo().browse(sale_order_id)
        else:
            return request.redirect('/shop')
```

Reply:
Nathan's suggestion is good, but as the orm call is made with a sudo, make sure to keep the "sale_order belongs to partner_id" verification.